src/server/src/main/java/io/cassandrareaper/core/Stream.java

```diff
@@ -0,0 +1,289 @@
+   */
+  private long sizeReceived;
+
+  private Stream(Builder builder) {
+    this.host = builder.host;
+    this.peer = builder.peer;
+    this.sizeToSend = builder.sizeToSend;
+    this.sizeToReceive = builder.sizeToReceive;
+    this.progressSent = builder.progressSent;
+    this.progressReceived = builder.progressReceived;
+    this.lastUpdated = builder.lastUpdated;
+    this.completed = builder.completed;
+    this.success = builder.success;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public String getPeer() {
+    return peer;
+  }
+
+  public long getSizeToSend() {
+    return sizeToSend;
+  }
+
+  public long getSizeToReceive() {
+    return sizeToReceive;
+  }
+
+  /**
+   * Computes number of bytes currently sent from host to peer.
+   * @return sum of bytes sent across all files in this stream.
+   */
+  public long getSizeSent() {
+    return this.progressSent.values().stream()
+        .map(fileProgress -> fileProgress.currentBytes)
+        .mapToLong(Long::longValue).sum();
+  }
+
+  /**
+   * Computes the number of bytes currently received by host from peer.
+   * @return sum of bytes received across all files in this stream.
+   */
+  public long getSizeReceived() {
+    return this.progressReceived.values().stream()
+        .map(fileProgress -> fileProgress.currentBytes)
+        .mapToLong(Long::longValue).sum();
+  }
+
+  public Map<String, FileProgress> getProgressReceived() {
+    return progressReceived;
+  }
+
+  public Map<String, FileProgress> getProgressSent() {
+    return progressSent;
+  }
+
+  public long getLastUpdated() {
+    return lastUpdated;
+  }
+
+  public boolean getCompleted() {
+    return completed;
+  }
+
+  public boolean getSuccess() {
+    return success;
+  }
+
+  public void setSizeSent(ProgressInfo progressInfo) {
+    progressSent.compute(progressInfo.fileName, (fileName, oldProgress) -> toFileProgress(progressInfo));
+  }
+
+  public void setSizeReceived(ProgressInfo progressInfo) {
+    progressReceived.compute(progressInfo.fileName, (fileName, oldProgress) -> toFileProgress(progressInfo));
+  }
+
+  public void setLastUpdated(long lastUpdated) {
+    this.lastUpdated = lastUpdated;
+  }
+
+  public void setCompleted(boolean completed) {
+    this.completed = completed;
+  }
+
+  public void setSuccess(boolean success) {
+    this.success = success;
+  }
+
+  public String streamId() {
+    return streamId(this.host, this.peer);
+  }
+
+  public static String streamId(String host, String peer) {
+    return String.format("%s->%s", host, peer.replaceAll("/", ""));
+  }
+
+  public String toString() {
+    return String.format(
+        "Stream(host=%s, peer=%s, toSend=%d, sent=%d, toReceive=%d, received=%d, updated=%d, completed=%b",
+        host, peer, sizeToSend, getSizeSent(), sizeToReceive, getSizeReceived(), lastUpdated, completed);
+  }
+
+  public static Stream.Builder builder() {
+    return new Stream.Builder();
+  }
```

Comment:
presuming `StreamFactory` was in the same package, this method and the following Builder class should be package-protected.